server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -128,7 +131,19 @@ public HttpInfo info() {
 
     @Override
     public HttpStats stats() {
-        return new HttpStats(httpChannels.size(), totalChannelsAccepted.get());
+        pruneClientStats();
+        return new HttpStats(new ArrayList<>(httpChannelStats.values()), httpChannels.size(), totalChannelsAccepted.get());
+    }
+
+    void pruneClientStats() {
+        // prune stale entries
+        long nowMillis = threadPool.absoluteTimeInMillis();
```

Comment:
@dakrone, thanks for reviewing this. I'm going to go through your comments and try to address them. With regard to this comment, the time values (e.g., open time, closed time, last request time) are all meant to be absolute. The only place a duration is calculated is here in the `pruneClientStats` method where any channels closed for longer than 5 minutes are pruned. I can add a separate nanosecond counter for the closed time if you think it's important that the pruning interval not be affected by time changes.

Reply:
okay, since they're absolute times I think it's okay, since we want them to correspond to real clock times (at least according to the machine ES is running on), thanks for the explanation!